### PR TITLE
Cpp activity exportation and structural fixes

### PR DESCRIPTION
**Pull request by nagyandras95**
This PR contains c++ activities exportation witch includes method bodies, effects, entry and exit actions and guard constraints. Furthermore it involved with some structural development like association or constructor exportation. It modifies the signals handling too. The Timer is partially handled but it needs more development. The multi-threading has been also modified. 
Those demo examples witch can be complied in the master can be complied with this branch too. The Airport model contains some not supported elements witch also requires reviewing UML export (like collection export..)
@gerazo, @devaigergely81 please, review it!

### CODE REVIEW

**Comment by gerazo** [comment]
retest this please

**Comment by gerazo** [comment]
Please remove all the committed .class files from this merge! There are too much and I can't see the real changes because of the committed binaries. They are all in bin directories which are (should be) protected by .gitignore files. Please remove the .class files and check that .gitignore rules are good. Tell me if you are ready and I can proceed with the review!

**Comment by gerazo** [comment]
Meanwhile, we also have a cpp compilation error. (File is missing)

**Comment by nagyandras95** [comment]
retest this please

**Comment by djnemeth** [comment]
@nagyandras95, some of your commits have invalid author email addresses. Please always make sure (before pushing) that you're committing using addresses associated with your GitHub account. See [your email settings](https://github.com/settings/emails) and the article '[Setting your email in Git](https://help.github.com/articles/setting-your-email-in-git/)'.

**Comment by djnemeth** [comment]
Jenkins, retest this, please.

**Comment by nagyandras95** [comment]
Retest this please.

**Comment by gerazo** [comment]
retest this please

**Comment by nagyandras95** [comment]
retest this please

**Comment by gerazo** [comment]
retest this please

File: dev/plugins/hu.elte.txtuml.export.cpp/cpp-runtime/assocation.hpp [hunk 1]
```diff
@@ -0,0 +1,85 @@
+#ifndef ASSOCATION_H
```
Comment (gerazo) [inline dev/plugins/hu.elte.txtuml.export.cpp/cpp-runtime/assocation.hpp]: assocation !=
association
also in file name

File: dev/plugins/hu.elte.txtuml.export.cpp/cpp-runtime/itimer.hpp [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+#include "statemachineI.hpp"
+#include "runtimetypes.hpp"
+class Timer;
+
+class ITimer
+{
+public:
+    static Timer* start(StateMachineI*,EventPtr,int);
+
+    virtual int quary() = 0;
```
Comment (gerazo) [inline dev/plugins/hu.elte.txtuml.export.cpp/cpp-runtime/itimer.hpp]: query != quary

File: dev/plugins/hu.elte.txtuml.export.cpp/cpp-runtime/timer.hpp [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+class Timer : public ITimer
+{           typedef std::chrono::milliseconds milliseconds;
+       public:
+
+           Timer(StateMachineI*,EventPtr,int);
+           ~Timer();
+
+           //TODO implement
+           virtual void reset(int) {}
+           virtual int quary() {return 0;}
```
Comment (gerazo) [inline dev/plugins/hu.elte.txtuml.export.cpp/cpp-runtime/timer.hpp]: querry

**Comment by gerazo** [comment]
It seems good at 1st sight.

File: dev/plugins/hu.elte.txtuml.export.cpp/cpp-runtime/itimer.hpp [hunk 4]
```diff
@@ -10,7 +10,7 @@ class ITimer
 public:
     static Timer* start(StateMachineI*,EventPtr,int);
 
-    virtual int quary() = 0;
+    virtual int query() = 0;
     virtual void reset(int) = 0;
     virtual void add(int) = 0;
     virtual bool canel() = 0;
```
Comment (djnemeth) [inline dev/plugins/hu.elte.txtuml.export.cpp/cpp-runtime/itimer.hpp]: @nagyandras95, isn't this supposed to be *cancel*?

**Comment by GregoBalu** [comment]
retest this please

**Comment by GregoBalu** [comment]
Seems OK to me code-wise and functionally.
I did not have the time to properly test it.

**Comment by nagyandras95** [comment]
retest this please

**Comment by nagyandras95** [comment]
Please, someone merge back this PR if no one has found any critical problem.

**Comment by djnemeth** [comment]
For some reason, C++ tests have been running for more than 2 hours now. Strangely, this happened to some earlier builds during the week as well, even outside this pull request.

**Comment by gerazo** [comment]
Hi. I've updated the server while tests were running. This could be the cause. Sorry.

**Comment by gerazo** [comment]
retest this please

**Comment by nagyandras95** [comment]
I have got more feedback that the PR should be merged back so I will do it.

**Pull request merged**